crates/evm-prover/src/storage/storage.rs

```diff
@@ -0,0 +1,265 @@
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use evm_exec_types::{BlockExecOutput, BlockRangeExecOutput};
+use rocksdb::{ColumnFamily, ColumnFamilyDescriptor, Options, DB};
+use serde::{Deserialize, Serialize};
+use sp1_sdk::SP1ProofWithPublicValues;
+use std::path::Path;
+use std::sync::Arc;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum ProofStorageError {
+    #[error("Database error: {0}")]
+    Database(#[from] rocksdb::Error),
+    #[error("Serialization error: {0}")]
+    Serialization(#[from] bincode::Error),
+    #[error("General error: {0}")]
+    General(#[from] anyhow::Error),
+    #[error("Proof not found for height: {0}")]
+    ProofNotFound(u64),
+    #[error("Range proof not found for range: {0}-{1}")]
+    #[allow(dead_code)]
+    RangeProofNotFound(u64, u64),
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct StoredBlockProof {
+    pub celestia_height: u64,
+    pub proof_data: Vec<u8>,
+    pub public_values: Vec<u8>,
+    pub created_at: u64,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct StoredRangeProof {
+    pub start_height: u64,
+    pub end_height: u64,
+    pub proof_data: Vec<u8>,
+    pub public_values: Vec<u8>,
+    pub created_at: u64,
+}
+
+#[async_trait]
+pub trait ProofStorage: Send + Sync {
+    async fn store_block_proof(
+        &self,
+        celestia_height: u64,
+        proof: &SP1ProofWithPublicValues,
+        output: &BlockExecOutput,
+    ) -> Result<(), ProofStorageError>;
+
+    #[allow(dead_code)]
+    async fn store_range_proof(
+        &self,
+        start_height: u64,
+        end_height: u64,
+        proof: &SP1ProofWithPublicValues,
+        output: &BlockRangeExecOutput,
+    ) -> Result<(), ProofStorageError>;
+
+    #[allow(dead_code)]
+    async fn get_block_proof(&self, celestia_height: u64) -> Result<StoredBlockProof, ProofStorageError>;
+
+    #[allow(dead_code)]
+    async fn get_range_proofs(
+        &self,
+        start_height: u64,
+        end_height: u64,
+    ) -> Result<Vec<StoredRangeProof>, ProofStorageError>;
+
+    #[allow(dead_code)]
+    async fn get_block_proofs_in_range(
+        &self,
+        start_height: u64,
+        end_height: u64,
+    ) -> Result<Vec<StoredBlockProof>, ProofStorageError>;
+
+    #[allow(dead_code)]
+    async fn get_latest_block_proof(&self) -> Result<Option<StoredBlockProof>, ProofStorageError>;
+}
+
+pub struct RocksDbProofStorage {
+    db: Arc<DB>,
+}
```

Comment:
When Jonas has finished his work creating the circuit state membership/inclusion we will need to integrate it into the prover service also. 

Should we be sharing a single `Arc<DB>` for all storage and have data arranged by column families (as you have already done here)?

I assume the approach would be to just add a new `CF_MESSAGES` for hyperlane messages, rather than creating another struct with a db field.